src/client/ua_client_connect.c

```diff
@@ -12,6 +12,20 @@
 
 #define UA_MINMESSAGESIZE 8192
 
+
+ /********************/
+ /* Set client state */
+ /********************/
+static void
+setClientState(UA_Client *client, UA_ClientState state)
+{
+    client->state = state;
```

Comment:
Can we simply check if state == client-state at the beginning?
This avoids having a second lastSendState variable.

Reply:
Yes ! 👍